### PR TITLE
Add 'libarchive' package 

### PR DESCRIPTION
**Pull request by tstack**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html) step by step carefully. **Yes**

* My change will resolve this "New package" request: https://github.com/cpp-pm/hunter/issues/182

---

### CODE REVIEW

File: docs/packages/pkg/libarchive.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   single: compression ; libarchive
+
+.. _pkg.libarchive:
+
+libarchive
+==========
+
+-  `Official <https://github.com/libarchive/libarchive>`__
+-  `Example <https://github.com/cpp-pm/hunter/blob/master/examples/libarchive/CMakeLists.txt>`__
+-  Added by `Timothy Stack <https://github.com/tstack>`__ (`pr-NNN <https://github.com/cpp-pm/hunter/pull/NNN>`__)
```
Comment (rbsheth) [inline docs/packages/pkg/libarchive.rst:16]: Please update PR number

**Review by rbsheth — COMMENTED** [review]
@tstack Thanks for this PR! It seems like upstream has a CMakeLists.txt file now - can we use that? Would be useful to get things working on Windows also.

https://github.com/libarchive/libarchive/blob/master/CMakeLists.txt

**Comment by tstack** [comment]
It sounded like the CMakeLists.txt would need to be forked/patched, which I'd rather not do.  So, I went the autotools route.  Has there been any progress on getting hunter working without needing to do forks of packages?

**Review by rbsheth — APPROVED** [review]
Okay, but we should fork/patch really for proper cross-platform support.

**Comment by rbsheth** [comment]
And no, we haven't implemented fork-free packages yet.

**Comment by rahulshethsc** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.280

**Pull request merged**